20240124_chwide-noise/chwide.py

```diff
@@ -0,0 +1,231 @@
+"""Module that produces simple noise maps for s4 Chile wide survey
+
+jcarron Jan 2024
+"""
+import glob
+import os
+import healpy as hp
+import numpy as np
+import numpy.random
+
+from lenspyx.utils_hp import Alm, almxfl, gauss_beam
+from lenspyx.remapping.utils_geom import Geom
+from lenspyx.utils import cli
+from scipy.interpolate import UnivariateSpline as spl
+from psutil import cpu_count
+
+from lenspyx.utils import timer
+from lenspyx import cachers
+
+UNSEEN = -1.6375e+30
+assert hp.UNSEEN == UNSEEN
+pix_min, npix = 14135296, 33839732
+
+# --- freqs on disk for alt1 chile wide ---
+freqs = [30, 40, 90, 150, 220, 280]
+XYs = ['II', 'IQ', 'IU', 'QQ', 'QU', 'UU']
+
+fns = glob.glob(os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/*.fits')
+if len(fns) == 2 * len(freqs): # Working locally
+    print("Using files on " + os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide')
+    path2cov = os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/cov_%03d.fits'  # in K^2, nside 128
+    path2hits = os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/hits_%03d.fits'
+else: # Most likely we are on NERSC
+    path2cov = '/global/cfs/cdirs/cmbs4/AoA/August2022/scaled_outputs/alternative_1/chlat_cd_wide/cov_%03d.fits' # in K^2, nside 128
+    path2hits = '/global/cfs/cdirs/cmbs4/AoA/August2022/scaled_outputs/alternative_1/chlat_cd_wide/hits_%03d.fits'
```

Comment:
we probably need to share also covariance matrices and hitmaps, right?

Reply:
@keskitalo is it ok if we share these files with the `s4xlitebird` group?

Reply:
I would rather work off the DC0 hitmaps and matrices but I suppose AoA will do in a pinch.  As for sharing files with s4xlitebird I presume the MOU specifies what information will be shared. If it doesn't, it may take an act of the science council to clear material for sharing.

Reply:
the MOU does not go down to this level of detail, but any data product that is necessary for s4xlitebird work can be shared with the project, so I am positive we can proceed to copy those data products into the s4xlitebird space. However let's wait until someone asks for them, maybe they are not needed.